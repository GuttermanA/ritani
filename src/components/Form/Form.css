.form.field {
  padding: 0.5rem 1rem;
  font-size: 1.1rem;
}

.form.button {
  background-color: #008cba;
  color: white;
  font-size: 1em;
  text-align: center;
  text-decoration: none;
  border: none;
  margin: 0.5rem;
  padding: 0.75rem 1.25rem;
}

.form.label {
  color: white;
  padding: 0.75rem 1.25rem;
  font-weight: bold;
}
