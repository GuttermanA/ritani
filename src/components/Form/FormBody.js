import React, { Component } from "React";

export default class FormBody extends Component {
  state = {};

  render() {
    const { children } = this.props;
    return <form>{children}</form>;
  }
}
