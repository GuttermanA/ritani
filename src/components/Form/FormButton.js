import React, { Component } from "react";
import PropTypes from "prop-types";
import Button from "../Button";

class FormButton extends Component {
  static propTypes = {
    as: PropTypes.elementType,
    action: PropTypes.string,
    children: PropTypes.node,
    className: PropTypes.string,
    disabled: PropTypes.bool,
    onClick: PropTypes.func
  };

  static defaultProps = {
    as: "button"
  };

  render() {
    // const ElementType = getElementType(FormButton, this.props);
    const { children, className, type = "submit", name, disabled } = this.props;
    return (
      <Button name={name} type={type} className={className} disabled={disabled}>
        {children}
      </Button>
    );
  }
}

export default FormButton;
