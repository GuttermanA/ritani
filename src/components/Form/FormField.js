import React, { Component } from "react";

class FormField extends Component {
  render() {
    const { className, type, name, required, placeholder } = this.props;
    return <input className type name required placeholder />;
  }
}

export default FormField;
