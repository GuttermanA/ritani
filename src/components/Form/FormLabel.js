import React, { Component } from "react";

class FormLabel extends Component {
  render() {
    const { children, fieldName, className = "form label" } = this.props;
    return (
      <label className={className} htmlFor={fieldName}>
        {children}
      </label>
    );
  }
}

export default FormLabel;
