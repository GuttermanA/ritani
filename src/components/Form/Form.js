import React, { Component } from "react";
import FormButton from "./FormButton";
import FormField from "./FormField";
import FormLabel from "./FormLabel";

export default class Form extends Component {
  static Button = FormButton;
  static Field = FormField;
  static FormLabel = FormLabel;

  state = {};

  handleSumbit = event => {
    alter("SUBMITTED");
    event.preventDefault();
  };

  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  render() {
    return <form onSumbit={this.handleSubmit}>{children}</form>;
  }
}
