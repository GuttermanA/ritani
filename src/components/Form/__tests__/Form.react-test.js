import React from "react";
import renderer from "react-test-renderer";
import Form from "../Form";

describe("<Form/>", () => {
  it("renders correctly", () => {
    const component = renderer.create(
      <Form>
        <Form.Field />
        <Form.Button />
      </Form>
    );
    expect(component).toMatchSnapshot();
  });
});
