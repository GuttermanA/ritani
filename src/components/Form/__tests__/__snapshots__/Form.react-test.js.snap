// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> renders correctly 1`] = `
<form
  as="form"
>
  <input
    className="form field"
    required={true}
    type="text"
  />
  <button
    className="form button"
    type="submit"
  />
</form>
`;
